thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiLog.java

```diff
@@ -99,6 +125,10 @@ public static void i(final String tag, final String msg) {
      * </code>
      */
     public static void setLogger(@Nullable final Logger logger) {
+        if (logger == TI_LOG) {
+            throw new IllegalArgumentException(
+                    "Recursion warning: You can't use TI_LOG as Logger for TiLog");
```

Comment:
It's a implementation of this interface which can be used everywhere except for this location which would result in recursion.
I really like that the `LoggingInterceptor` logs to `TiLog` by default using the `TiLog.TI_LOG`.